partiql-eval/src/eval.rs

```diff
@@ -949,6 +949,7 @@ impl EvalExpr for EvalBinOpExpr {
                 (EvalBinOp::Or, Boolean(true)) => Some(true.into()),
                 (EvalBinOp::Or, Null) | (EvalBinOp::Or, Missing) => Some(Null),
                 (EvalBinOp::In, Null) | (EvalBinOp::In, Missing) => Some(Null),
+                (EvalBinOp::Concat, Null) | (EvalBinOp::Concat, Missing) => Some(Null),
```

Comment:
I don't think concat should be propagating `NULL` over `MISSING`. Based on the [conformance tests for concat](https://github.com/partiql/partiql-tests/blob/main/partiql-tests-data/eval/primitives/operators/concat.ion#L87-L145), `MISSING` is propagated over `NULL`. I think only the logical operators (i.e. `AND`, `OR`, `IN`) will propagate `NULL` over `MISSING`.